python/TestHarness/testers/RunApp.py

```diff
@@ -122,6 +123,10 @@ def getCommand(self, options):
     if options.scaling and specs['scale_refine'] > 0:
       specs['cli_args'].insert(0, ' -r ' + str(specs['scale_refine']))
 
+    # TestHarness default is to NOT print errors from all ranks on parallel tests
+    if not specs['keep_cerr']:
+      specs['cli_args'].append('--drop-cerr')
```

Comment:
This is fine except I think I'd rather have it in the `RunException` class. The reason being is I don't want to suppress errors for the normal case just in case somebody throws only on a non-zero rank and gets confused because the output gets dropped. 

Alternatively we could leave this here but change the default to true for `RunApp` but set it to false for `RunException`.